1_prep/src/munge_meteo.R

```diff
@@ -0,0 +1,66 @@
+#' @title Split each GCM netCDF into feather files specific to cells and time periods
+#' @decription Create a feather file with driver data for each gcm, 
+#' for each cell, for each time period
+#' @param gcm_nc filenames of GCM netCDF files
+#' @param gcm_name name of one of the six GCMs
+#' @param cell_id id of a single GCM cell that is known to contain lakes
+#' @param gcm_date bracketing years of a single GCM time period
+#' @param outfile_template string representing the filepath at which to save
+#' the feather file output of the split netCDF data. The first `%s` is
+#' used as the placeholder for the `gcm_name`, the second is for the `gcm_date`,
+#' and the third is for the `cell_no`.
+#' @return names of cell/gcm/time-period specific feather files
+munge_nc_files <- function(gcm_nc, gcm_name, cell_no, gcm_date, outfile_template) {
+  # open netCDF file (gcm-specific) and pull data for that cell, for that time period
+  # TODO - actually pull time data for only one time period - pulling all for now
+  nc <- nc_open(gcm_nc)
+  time_dim <- ncvar_get(nc, "time")
+  cell_dim <- ncvar_get(nc, "gridcells")
+  data <- purrr::map_df(nc$var, function(var) {
+    ncvar_get(nc, var, start = c(1,cell_no), count = c(-1,1))
+  }) %>% mutate(time = nc.get.time.series(f = nc,time.dim.name = "time"),.before=1)
+
+  
+  # write to feather file, specific to gcm, cell, and time period
+  outfile <- sprintf(outfile_template, gcm_name, gcm_date, cell_no)
+  if(!dir.exists(dirname(outfile))) dir.create(dirname(outfile), recursive=TRUE)
+  arrow::write_feather(data, outfile)
+  
+  return(outfile)
+}
+
+#' @title build a lake-gcm-time-period crosswalk table
+#' @description Build a dplyr table with tar_grouping that has one row per
+#' Lake-GCM-time-period combination and columns for the (1) tar branch name 
+#' and (2) data hash for the meteo data for each combo. We can use this to run models.
+#' For the string extraction to work, cell_no must be at the END of the meteo filename
+#' @param meteo_feathers - names of the cell/gcm/time-period specific feather files
+#' @param lake_cell_xwalk - mapping of which lakes fall into which gcm cells
+#' @param gcm_names - names of the 6 GCMs
+#' @param gcm_dates - the three GCM time periods, defined by their bracketing years
+#' @return a dplyr tibble with one row per model run
+build_meteo_xwalk <- function(meteo_feathers, lake_cell_xwalk, gcm_names, gcm_dates){
```

Comment:
There is a part of this that feels like a `xwalk` but ultimately this is a configuration table for all of the model runs - it sets up the info needed by `run_glm3_model()` for each run. I wonder if there is a better name that makes this clear every time you see it mentioned? `build_meteo_config()` ? `build_meteo_model_run()`. Maybe not worth the effort to go change, though.

Reply:
Yeah I agree it's not a great name. Maybe I'll switch to `build_model_config()` and `p1_model_config`?